File: michielstokman/ai-implementation-fastapi/app/moderation.py
import re

BLOCKED_PATTERNS = [
    re.compile(r'self-harm', re.I),
    re.compile(r'suicide', re.I),
    re.compile(r'medical advice', re.I),
    re.compile(r'hate speech', re.I),
]


def moderate_content(text: str) -> dict:
    issues = [p.pattern for p in BLOCKED_PATTERNS if p.search(text or '')]
    return {'approved': len(issues) == 0, 'issues': issues}

File: michielstokman/ai-implementation/src/moderation.js
const blockedPatterns = [
  /self-harm/i,
  /suicide/i,
  /medical advice/i,
  /hate speech/i,
];

function moderateContent(text) {
  const issues = blockedPatterns.filter((rx) => rx.test(text)).map((rx) => rx.toString());
  return {
    approved: issues.length === 0,
    issues,
  };
}

module.exports = { moderateContent };
